crates/rpc-types-mev/src/eth_calls.rs

```diff
@@ -18,6 +18,9 @@ pub struct EthCallBundle {
     pub block_number: u64,
     /// Either a hex encoded number or a block tag for which state to base this simulation on
     pub state_block_number: BlockNumberOrTag,
+    /// The block number index to use for which state to base this simulation on (optional - overrides state_block_number)
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub block_number_index: Option<u64>,
```

Comment:
unclear to me what `index` refers to here, is this the transaction index in the block?

Reply:
yep is the transaction index you will simulate from

Reply:
Say you wanted to simulate the bundle from index 3 in the block, you would need to update the top of block state with transactions at index 0, 1 & 2 to get the state aligned with that index

Reply:
variety of options

tx_index
state_block_index
state_block_position
block_position

Let me know what makes most sense

Reply:
so this is basically eth_callMany's:

https://github.com/alloy-rs/alloy/blob/67576f9f6e70026116443851a43e4c805902c3b3/crates/rpc-types-eth/src/call.rs#L35-L38

could we use the same instead?